packages/sdk-socket-server-next/src/analytics-api.ts

```diff
@@ -314,22 +314,30 @@ app.post('/evt', async (_req, res) => {
       },
     };
 
-    // Replace 'sdk' id wichch translates to '5a374dcd2e5eb762b527af3a5bab6072a4d24493' with a unique random id
-    if (event.userId === SDK_EXTENSION_DEFAULT_ID) {
+    // Always check for userId to avoid hot sharding events
+    if (!event.userId) {
       const newUserId = uuidv4();
-      logger.info(`Replacing 'sdk' id with a random uuid=${newUserId}`);
+      logger.info(
+        `event: ${event.event} - Replacing 'sdk' id with '${newUserId}'`,
+        event,
+      );
       event.userId = newUserId;
-      event.properties.userId = newUserId;
     }
 
     // Make sure each events have a valid dappId
-    if (!event.properties.dappId) {
-      logger.error(
-        `event: ${event.event} - dappId is required - event will be ignored`,
+    // Replace 'sdk' id which translates to '5a374dcd2e5eb762b527af3a5bab6072a4d24493' with fallback to url / title / random uuid
```

Comment:
here we are not targeting the same 'sdk' hotsharding.
Here our problem is that this is empty and therefore it's getting thrown away.
Would like to fill this in with as relevant info as we can. Otherwise we can leave it empty.
No need to fill that with a UUIDv4, as that is not adding any extra value.

Reply:
empty string will be detected as missing

Reply:
yes, and I think that is fine, as long as we don't throw out the event, but we send it to segment with an empty string.
I.e if we don't have anything valuable to put in that field server side, we should log the platform and version so we make sure we have it fixed client side. but otherwise not loose the event.

Reply:
the issue here is that if url is "N/A" we will send as N/A and can potentially be hotsharded

Reply:
I updated accordingly